tests/std/tests/P0586R2_integer_comparison/test.cpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//
+// Tests the new functions added as part of P0586R2, "Integer Comparison Functions"
+//
+
+#include <assert.h>
+#include <limits>
+#include <utility>
+
+using namespace std;
+
+constexpr bool test() {
+    assert(cmp_equal(1L, 1));
+    assert(!cmp_equal(1, -1));
+    assert(!cmp_equal(1, 42));
+    assert(!cmp_equal(42, 1));
+
+    assert(!cmp_not_equal(1L, 1));
+    assert(cmp_not_equal(1, -1));
+    assert(cmp_not_equal(1, 42));
+    assert(cmp_not_equal(42, 1));
+
+    assert(!cmp_less(1L, 1));
+    assert(!cmp_less(1, -1));
+    assert(cmp_less(1, 42));
+    assert(!cmp_less(42, 1));
+
+    assert(cmp_less_equal(1L, 1));
+    assert(!cmp_less_equal(1, -1));
+    assert(cmp_less_equal(1, 42));
+    assert(!cmp_less_equal(42, 1));
+
+    assert(!cmp_greater(1L, 1));
+    assert(cmp_greater(1, -1));
+    assert(!cmp_greater(1, 42));
+    assert(cmp_greater(42, 1));
+
+    assert(cmp_greater_equal(1L, 1));
+    assert(cmp_greater_equal(1, -1));
+    assert(!cmp_greater_equal(1, 42));
+    assert(cmp_greater_equal(42, 1));
+
+    assert(in_range<int>(1));
+    assert(in_range<int>(1L));
+    assert(!in_range<unsigned int>(-1));
+    assert(in_range<unsigned int>(0));
+    assert(in_range<unsigned int>(42));
+    assert(!in_range<int>(numeric_limits<long long>::max()));
+    assert(!in_range<int>(numeric_limits<long long>::min()));
+    assert(in_range<int>(numeric_limits<int>::max()));
+    assert(in_range<int>(numeric_limits<int>::min()));
+    assert(in_range<long>(numeric_limits<int>::max()));
+    assert(in_range<long>(numeric_limits<int>::min()));
+
+    { // Different signs
+        const int i          = -1;
+        const unsigned int u = 1;

```

Comment:
I'd like to see some `make_signed_t<T>{-1}, make_unsigned_t<T>(-1)` test cases as well.